### PR TITLE
TDI-43477 : fix credentials error message

### PR DESCRIPTION
**Pull request by rlecomte-talend**

### CODE REVIEW

File: bigquery/src/main/java/org/talend/components/bigquery/service/BigQueryService.java [hunk 1]
```diff
@@ -430,13 +430,12 @@ private Gson createGsonBuilder() {
...
-    public static GoogleCredentials getCredentials(String credentials) {
+    public GoogleCredentials getCredentials(String credentials) {
         try {
             return GoogleCredentials.fromStream(new ByteArrayInputStream(credentials.getBytes()))
                     .createScoped(BigqueryScopes.all());
-        } catch (IOException e) {
-            throw new BigQueryConnectorException(
-                    "Exception when read service account file: " + credentials + "\nMessage is:" + e.getMessage());
+        } catch (Exception e) {
+            throw new BigQueryConnectorException(i18n.errorReadingCredentials() + " : " + e.getMessage());
```
Comment (ypiel-talend) [inline bigquery/src/main/java/org/talend/components/bigquery/service/BigQueryService.java]: Add ":" in the Message.properties message. So I18n team will understand why there is no ending dot.

Comment (rlecomte-talend) [reply bigquery/src/main/java/org/talend/components/bigquery/service/BigQueryService.java]: done

File: bigquery/src/main/java/org/talend/components/bigquery/service/BigQueryService.java [hunk 2]
```diff
@@ -430,13 +430,12 @@ private Gson createGsonBuilder() {
...
-    public static GoogleCredentials getCredentials(String credentials) {
+    public GoogleCredentials getCredentials(String credentials) {
         try {
             return GoogleCredentials.fromStream(new ByteArrayInputStream(credentials.getBytes()))
                     .createScoped(BigqueryScopes.all());
-        } catch (IOException e) {
-            throw new BigQueryConnectorException(
-                    "Exception when read service account file: " + credentials + "\nMessage is:" + e.getMessage());
+        } catch (Exception e) {
+            throw new BigQueryConnectorException(i18n.errorReadingCredentials() + e.getMessage());
```
Comment (clesaec) [inline bigquery/src/main/java/org/talend/components/bigquery/service/BigQueryService.java]: Better to include exception message in errorReading message :
`throw new BigQueryConnectorException(i18n.errorReadingCredentials(e.getMessage()));`
and you can put cause exception in BigQueryConnectorException
`throw new BigQueryConnectorException(i18n.errorReadingCredentials(e.getMessage()), e);`

File: bigquery/src/main/java/org/talend/components/bigquery/service/I18nMessage.java [hunk 3]
```diff
@@ -52,4 +52,6 @@
     String projectNameRequired();
 
     String credentialsRequired();
+
+    String errorReadingCredentials();
```
Comment (clesaec) [inline bigquery/src/main/java/org/talend/components/bigquery/service/I18nMessage.java]: as commented upper :
`String errorReadingCredentials(String expMsg);`

File: bigquery/src/main/resources/org/talend/components/bigquery/service/Messages.properties [hunk 4]
```diff
@@ -7,6 +7,7 @@ org.talend.components.bigquery.service.I18nMessage.tableOperationIsNotSupported
 org.talend.components.bigquery.service.I18nMessage.writeOperationIsNotSupported = Write operation {0} is not defined
 org.talend.components.bigquery.service.I18nMessage.projectNameRequired = Project name is required
 org.talend.components.bigquery.service.I18nMessage.credentialsRequired = Google credentials are required
+org.talend.components.bigquery.service.I18nMessage.errorReadingCredentials = Error while reading credentials :
```
Comment (clesaec) [inline bigquery/src/main/resources/org/talend/components/bigquery/service/Messages.properties]: `Error while reading credentials : {0}` as well

Comment (rlecomte-talend) [reply bigquery/src/main/resources/org/talend/components/bigquery/service/Messages.properties]: done

**Pull request merged**